dspace-server-webapp/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java

```diff
@@ -37,12 +46,22 @@
 
     private static final Logger log = LoggerFactory.getLogger(StatelessAuthenticationFilter.class);
 
+    private static final String ON_BEHALF_OF_REQUEST_PARAM = "X-On-Behalf-Of";
+
     private RestAuthenticationService restAuthenticationService;
 
     private EPersonRestAuthenticationProvider authenticationProvider;
 
     private RequestService requestService;
 
+    private AuthorizeService authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+
+    private EPersonService ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
+
+    private ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+
+    private boolean inErrorOnBehalfOf = false;
```

Comment:
this cannot be an instance attribute otherwise there are issue when multiple login request are processed at the same time